Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/revert.py

```diff
@@ -73,23 +73,61 @@ def parser(cls, parser, loggers=None):
         )
 
     @classmethod
-    def get_issue_info(cls, args, repository, **kwargs):
-        # Can give either a bug URL or a title of the new issue.
+    def get_commit_info(cls, args, repository, **kwargs):
+        commit_objects = []
+        commits_to_revert = []
+        commit_bugs = []
+        for c in args.commit:
+            try:
+                commit = repository.find(c, include_log=True)
+            except (local.Scm.Exception, ValueError) as exception:
+                sys.stderr.write('Could not find "{}"'.format(c) + '\n')
+                return None, None, None
+            commit_objects.append(commit)
+            commits_to_revert.append(commit.hash)
+            commit_bugs += [Tracker.from_string(i.link) for i in commit.issues if isinstance(Tracker.from_string(i.link).tracker, bugzilla.Tracker)]
+        return commit_objects, commits_to_revert, commit_bugs
+
+    @classmethod
+    def get_issue_info(cls, args, repository, commit_objects, commit_bugs, **kwargs):
+        # Can give either a bug URL or a title of the new issue
         # Using --revert-issue to differentiate from PR arg --issue/-i
         if not args.revert_issue:
-            prompt = 'Enter issue URL or title of new issue for the revert: '
+            prompt = 'Enter issue URL or reason for the revert: '
             args.issue = Terminal.input(prompt, alert_after=2 * Terminal.RING_INTERVAL)
         else:
             args.issue = args.revert_issue
-
         issue = Tracker.from_string(args.issue)
 
         if not issue and Tracker.instance() and getattr(args, 'update_issue', True):
             if getattr(Tracker.instance(), 'credentials', None):
                 Tracker.instance().credentials(required=True, validate=True)
+
+            # Automatically set project, component, and version of new bug
+            print('Setting bug properties...')
+            commit_bug = commit_bugs.pop()
+            project = commit_bug.project
+            component = commit_bug.component
+            version = commit_bug.version
+
+            # Check whether properties are the same if multiple commits are reverted
+            while len(commit_bugs):
+                commit_bug = commit_bugs.pop()
+                # Setting properties to None will prompt for user input
+                if commit_bug.project != project:
+                    project = None
+                if commit_bug.component != component:
+                    component = None
+                if commit_bug.version != version:
+                    version = None
+
+            commits = [repr(c) for c in commit_objects]
             issue = Tracker.instance().create(
                 title=args.issue,
-                description=Terminal.input('Issue description: '),
+                description='Revert {} because {}.'.format(string_utils.join(commits), args.issue),
```

Comment:
See the discussion of "issue" vs "why a few lines up.

Reply:
Oops, I missed this in my recent changes. Is there ever a world where someone looks for an issue using its title? If not, I think I can change the code such that we use `args.reason` here to make it clearer.